README.md

```diff
@@ -198,114 +198,156 @@ Mulang is just a Haskell library. You can install it though cabal.
+   "spec" : {
+      "expectations" : [
+         {
+            "subject" : [
+               "x"
+            ],
+            "object" : {
+               "contents" : [],
+               "tag" : "Anyone"
+            },
+            "negated" : false,
+            "verb" : "uses",
+            "transitive" : false,
+            "tag" : "Advanced"
+         }
+      ],
+      "signatureAnalysisType" : {
+        "tag" : "NoSignatures",
+        "contents" : []
+      }
+   }
+}
+' | json_pp
 {
-   "results" : [
+   "signatures" : [],
+   "tag" : "AnalysisCompleted",
+   "smells" : [],
+   "expectationResults" : [
       {
          "expectation" : {
-            "tag" : "Advanced",
             "object" : {
                "tag" : "Anyone",
                "contents" : []
             },
-            "verb" : "uses",
             "transitive" : false,
             "negated" : false,
             "subject" : [
                "x"
-            ]
+            ],
+            "verb" : "uses",
+            "tag" : "Advanced"
          },
          "result" : false
       }
-   ],
-   "smells" : [],
-   "signatures" : []
+   ]
 }
 ```
 
-...with basic expectations:
+### With basic expectations
 
 ```bash
 $ mulang '
 {
-   "code" : {
-      "content" : "x = 1",
-      "language" : "Haskell"
+   "sample" : {
+      "language" : "Haskell",
+      "content" : "x = 1"
    },
-   "analyseSignatures" : false,
-   "expectations" : [
-      {
-         "tag" : "Basic",
-         "inspection" : "HasBinding",
-         "binding" : "x"
-      }
-   ]
-}' | json_pp
+   "spec" : {
+      "signatureAnalysisType" : {
+        "tag" : "NoSignatures",
+        "contents" : []
+      },
+      "expectations" : [
+         {
+            "binding" : "x",
+            "tag" : "Basic",
+            "inspection" : "HasBinding"
+         }
+      ]
+   }
+}
+' | json_pp
 {
+   "tag" : "AnalysisCompleted",
    "signatures" : [],
    "smells" : [],
-   "results" : [
+   "expectationResults" : [
       {
-         "result" : true,
          "expectation" : {
+            "tag" : "Basic",
             "inspection" : "HasBinding",
-            "binding" : "x",
-            "tag" : "Basic"
-         }
+            "binding" : "x"
+         },
+         "result" : true
       }
    ]
 }
 
 ```
 
-...and with signature analysis:
+### With signature analysis
 
 ```bash
 $ mulang '
 {
-   "code" : {
-      "content" : "function foo(x, y) { return x + y; }",
-      "language" : "JavaScript"
+   "sample" : {
+      "language" : "JavaScript",
+      "content" : "function foo(x, y) { return x + y; }"
    },
-   "expectations" : [],
-   "analyseSignatures" : true
+   "spec" : {
+      "expectations" : [],
+      "signatureAnalysisType" : {
+        "tag" : "StyledSignatures",
+        "contents":"HaskellStyle"
+      }
+   }
 }' | json_pp
 {
-   "results" : [],
-   "smells" : [],
+   "expectationResults" : [],
    "signatures" : [
-      "foo(x, y)"
-   ]
+      "-- foo(x, y)"
```

Comment:
Because a signature is not valid code. Being a comment you can embed it in code, e.g. add it on top of original definition. 

From a teaching point of view, it also prevents students to think that signatures are a valid language construction.